src/data_interface.h

```diff
@@ -11,9 +11,18 @@ namespace zadeh {
 template<typename ArrayType, typename ElementType, typename IndexType = size_t>
 ElementType get_at(const ArrayType &candidates, const IndexType j);
 
+template<typename ArrayType, typename ElementType, typename IndexType = size_t>
+ElementType create(const ArrayType &candidates, const IndexType j);
+
 template<typename ArrayType, typename SizeType = size_t>
 SizeType get_size(const ArrayType &candidates);
 
+template<typename ArrayType, typename SizeType = size_t, typename AllocatorType>
+ArrayType init(const SizeType len, const optional<AllocatorType> &alloc = optional<AllocatorType>{});
```

Comment:
The second argument should be optional (for example we should use `std::allocator` by default for vectors). This `alloc` is mostly needed for `Napi::Array` and `Napi::Object` which we should pass `info::Env()` to.

Reply:
@jeff-hykin Do you have any idea how I should proceed with the second optional parameter in this generic declaration of the template? I used `optional`, but there are other ways too.